vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,1202 @@
+---
+title: "Comparing stringr to base R"
+author: "Sara Stoudt"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette aims to compare stringr functions to base R equivalents. Following the groupings of functions from the [stringr cheat sheet](https://github.com/rstudio/cheatsheets/blob/master/strings.pdf), these are the base equivalents to stringr functions.
+
+# Detect Matches
+
+| base | stringr    |
+|------|------------|
+|   `grepl()`   | `str_detect()` |
+|   `grep()`   | `str_which()`  |
+|   `gregexpr()`   | `str_count()`  |
+|  `gregexpr()`    | `str_locate()` |
+
+# Subset Strings
+
+| base | stringr    |
+|------|------------|
+|   `substr()`   | `str_sub()`    |
+|    `grep()`  | `str_subset()` |
+|   `regexpr()` + `regmatches()`   | `str_extract()`|
+|   `regexec()` + `regmatches()`   | `str_match()`  |
+
+
+# Manage Lengths
+
+| base | stringr    |
+|------|------------|
+|   `nchar()`   | `str_length()` |
+|  `sprintf()`    | `str_pad()`    |
+|   `substr()`   | `str_trunc()`  |
+|   `trimws()`   | `str_trim()`   |
+
+# Mutate Strings
+
+| base | stringr        |
+|------|----------------|
+|  `substr()`    | `str_sub()`        |
+|   `sub()`   | `str_replace()`    |
+|   `gsub()`   | `str_replace_all()`|
+|   `tolower()`   | `str_to_lower()`   |
+|   `toupper()`   | `str_to_upper()`   |
+|  `tools::toTitleCase()`    | `str_to_title()`   |
+
+
+# Join and Split
+
+| base | stringr        |
+|------|----------------|
+|   `paste()`   | `str_c()`          |
+|  `replicate()` + `paste()`    | `str_dup()`        |
+|  `strsplit()`    | `str_split_fixed()`|
+|   `paste()` + `eval()`   | `str_glue()`       |
+|    `paste()`  | `str_glue_data()`  |
+
+# Order Strings
+
+| base | stringr        |
+|------|----------------|
+|   `order()`   | `str_order()`      |
+|   `sort()`   | `str_sort()`       |
+
+# Helpers
+
+| base | stringr      |
+|------|--------------|
+|  `iconv()`    | `str_conv()`     |
+|  N/A    | `str_view()`     |
+|   N/A   | `str_view_all()` |
+|  `strwrap()`    | `str_wrap()`     |
+
+```{r}
+library(stringr)
+```
+
+The following examples walk through the examples in the stringr documentation and reproduce the results in base R.
```

Comment:
I think this needs a top-level heading, but I'm not entirely sure what the purpose is. Currently it seems like the goal is to allow people to replace stringr with base, which seems like a bit of a weird goal.

It's also rather long, and that along with the lack of a clear organising principle makes it a bit hard to review. So I think we should discuss here a bit and then come up with a plan for revisions.

Reply:
Thanks for the feedback! I'll go back and make the changes for all of your suggestions, and then I am happy to make a plan for larger scale revisions.

The vignette is certainly a bit unwieldy, and I can definitely cut and rearrange based on what you think makes sense. My original purpose was to show how gnarly some of the base R equivalents to simple stringr calls are, but I can see how heavy that ends up being.

I could organize the functions by the chunks in the tables above instead of in alphabetical order to make the vignette easier to navigate. Perhaps it is enough to show the difference in output of comparable functions to make it clear how un-tidy the output of base R functions can be (as opposed to going through the effort to make the outputs look identical). 

I also like your idea of emphasizing the difference in expected order of inputs between base and stringr.

Reply:
Random thought: would it be better to move the individual base R vs string examples into the function documentation?

Reply:
But first, maybe it's making sure we're clear on the goal. Is it to:

* help people transition from base to stringr?
* promote stringr by showing specific use cases?
* help people transition from stringr to base if they want to remove a package dependency?

I think the first is probably the most important for me.

Reply:
Your preferred goal makes sense; that is probably the biggest use case.

How about I plan for the next draft to be organized by functionality (detect matches, subset strings, manage lengths, mutate strings, order strings, join, and split), start with the base R example (minus the extra effort to tidy the output), and then introduce the equivalent stringr function (noting the difference in input order and the differences in output formatting)? I can also just pick the most representative example(s) for each function rather than doing each example provided in the documentation. This should streamline the vignette and make it more easily navigable. 

Outlining Example:

Suppose you want to know whether each word in a vector of fruit names contains an "a". In base you would use `grepl(pattern = "a", x = fruit)`. In stringr, the pattern comes last: `str_detect(x = fruit, pattern = "a")`. [maybe make a point about how this order of inputs is helpful for `apply()` and `purrr::map()` statements?]

Then move on to `grep()` and note that it can be manipulated to do various things, but stringr compartmentalizes functionality more (i.e. `grep()` can be used for both `str_which()` and `str_subset()`).

etc...

Then maybe at the end, note some common themes in switching from base R functions to stringr functions overall.

Reply:
Yeah, that sounds great!